src/main/java/tracko/model/items/Quantity.java

```diff
@@ -10,20 +10,20 @@ public class Quantity {
     public static final String MESSAGE_CONSTRAINTS =
             "Quantity should not be negative.";
 
-    public final int quantity;
+    public final Integer quantity;
 
     /**
      * Constructs a {@code Quantity}.
      * @param quantity The quantity of an item.
      */
-    public Quantity(int quantity) {
+    public Quantity(Integer quantity) {
         requireAllNonNull(quantity);
         checkArgument(isValidQuantity(quantity), MESSAGE_CONSTRAINTS);
         this.quantity = quantity;
     }
 
-    public static boolean isValidQuantity(int test) {
-        return !(test < 0);
+    public static boolean isValidQuantity(Integer test) {
+        return !(test < 0) && (test <= Integer.MAX_VALUE);
     }
```

Comment:
Correct me if I'm wrong, but I think Java throws an error here if the input Integer test is > Integer.MAX_VALUE, so it makes the second predicate here redundant. I tried it on an online REPL, since I didn't see any test statements to test the values more than Integer.MAX_VALUE. Let me know if I am wrong!

Reply:
Actually I'm pretty conflicted on whether to put the (test <= Integer.MAX_VALUE) at first, but after searching a bit I do think its redundant as well, bc apparently `Integer.MAX_VALUE + 1` is equal to `Integer.MIN_VALUE` (2's complement rule). Thanks for pointing this out!